### ISSUE TITLE
Feature Recommendation: Move InApp Message 'read' data from local storage to API

### ISSUE THREAD

**Issue by jehartzog** [Iterable/react-native-sdk#92]
## TL/DR

The `read` field on all InApp messages is stored locally on a single device, so this field is reset if a user logs out, and then logs back in, or switches device. We had expected this to be stored on the API, to ensure a user doesn't seen old messages already seen, even if they switch devices or have to re-auth.

## Long version

We recently integrated the InApp messages to replace some of our custom built announcement features. Overall the integration went very smoothly, but there was one case where the iOS and Android sdks (and by extension, this one) worked counter to how we expected, and how the documentation implies it works.

Looking at this statement from the [docs](https://support.iterable.com/hc/en-us/articles/217517406-In-App-Messages-and-Mobile-inbox-):
> Iterable will not show the same in-app message to the same user more than once unless the message has been re-sent or selected in a mobile inbox.

There is a huge exception to this that wasn't clear until we completed our integration and did testing. 

When a message is `consumed`, the SDK's update the API ([android](https://github.com/Iterable/iterable-android-sdk/blob/f8a459722a688ee1cc5eb1bb4fddaf4f908b746f/iterableapi/src/main/java/com/iterable/iterableapi/IterableInAppManager.java#L256), [ios](https://github.com/Iterable/swift-sdk/blob/dd27e72e4198fbf8ea28aeed0cb5332eb121ffec/swift-sdk/Internal/InAppManager.swift#L486))

When a message is `read`, no API call is made ([android](https://github.com/Iterable/iterable-android-sdk/blob/f8a459722a688ee1cc5eb1bb4fddaf4f908b746f/iterableapi/src/main/java/com/iterable/iterableapi/IterableInAppManager.java#L133), [ios](https://github.com/Iterable/swift-sdk/blob/dd27e72e4198fbf8ea28aeed0cb5332eb121ffec/swift-sdk/Internal/InAppManager.swift#L142)).

[These docs](https://support.iterable.com/hc/en-us/articles/360018709631#displaying-in-app-messages) indicate that the only API calls made by the Iterable SDK when a Message is read is to track `In-App Open` event, which seems to indicate this was a deliberate decision to not create an API endpoint for storing this `read` data field.

## Effort on our use case

This means if we leave messages in the state of `not consumed` but `read` sitting in the message inbox, if the user re-auths, switches devices, then they will re-receive all those messages as they will all be immediately reset to `read = false`.

## Recommendation

It would be nice to have this `read` field stored on the API, rather than locally on the device, so that this state is persisted through app re-installs or device switches.

**Comment by jehartzog** [comment]
I realize this may not be correct place to post this. Technically this would be a change in both iOS and Android SDKs, as well as API.

**Comment by jehartzog** [comment]
Just found the place in [api docs](https://api.iterable.com/api/docs#In-app_getMessages) which explicitly state this:
>This endpoint returns new messages and messages that have already been saved to a user’s mobile inbox. Since a message’s read status is local to a mobile device, there is not currently a way to differentiate between the two.

**Comment by roninopf** [comment]
We've now fully released this on every platform, so we'll close this out. Of course, if there's a problem, please feel free to reach out through your CSM and/or here!
